pandas/io/formats/info.py

```diff
@@ -158,17 +79,23 @@ def info(
     lines.append(str(type(data)))
     lines.append(data.index._summary())
 
-    if len(data.columns) == 0:
+    if data._typ == "dataframe":
+        cols = data.columns
+        dtypes = data.dtypes
+    else:
+        cols = Index([data.name])
+        dtypes = Index([data.dtypes])
+
+    col_count = len(cols)
```

Comment:
Hmm I think this is confusing to have `col_count` be populated by a Series in some cases - can you think of a way to refactor?

Reply:
@WillAyd sure, have given this a go